### DISCUSSION TITLE
Working with S3

### DISCUSSION THREAD

**Discussion by legaljustus** [Budibase/budibase#7432] [Help] [answered]
Hi!

Im trying to work with AWS S3 buckets in budibase. Currently the only field you can specifiy in the Query is the bucket name. And on making a request it just return everything. 

Isn't there a better way to do this or am i using this connector wrong? For example if i have the filename / key stored in a postgres database, is there no way to just request that specific file based on the path / filename?

Thank you in advance for any help!

**Comment by melohagan** [comment, marked as answer]
Hey @legaljustus 

Unfortunately you are correct that you can only return a list of objects by bucket name. 

I've raised a feature request here to expand upon the connector to allow for your use case: https://github.com/Budibase/budibase/issues/7451
